cmd/juju/caas/add.go

```diff
@@ -126,13 +127,19 @@ func (c *AddCAASCommand) Init(args []string) (err error) {
 
 // getStdinPipe returns nil if the context's stdin is not a pipe.
 func getStdinPipe(ctxt *cmd.Context) (io.Reader, error) {
-	if stdIn, ok := ctxt.Stdin.(*os.File); ok {
+	if stdIn, ok := ctxt.Stdin.(*os.File); ok && !terminal.IsTerminal(int(stdIn.Fd())) {
+		// stdIn from pipe but not terminal
 		stat, err := stdIn.Stat()
 		if err != nil {
 			return nil, err
 		}
-		if stat.Mode()&os.ModeNamedPipe != 0 {
-			return stdIn, nil
+		content, err := ioutil.ReadAll(stdIn)
```

Comment:
I do not like this as well (read to get size of the content), but Stat.Size() is always 0 for piped stdIn